src/main/java/com/example/pokebattlez/battle/controller/Battle.java

```diff
@@ -130,12 +138,18 @@ private int calculateDamage(Pokemon attacker, Pokemon defender, Pokemon.Move mov
     }
 
     private void playTurn() {
-        turnLog = new ArrayList<>();
-
-        playerActions.stream().sorted(Comparator.comparingInt(action -> action.priority)).forEachOrdered(playerAction -> playerAction.run(this));
+        playerActions.stream().sorted(Comparator.comparingInt(action -> action.priority))
+                .forEachOrdered(playerAction -> {
+                    playerAction.run(this);
```

Comment:
playerAction.run(_battle_) ? It's hard to tell what this really does. Can player's action run the battle?

Reply:
Had to pass it a Battle instance since the methods called by run() are static.